libs/widget-lib/src/types.ts

```diff
@@ -47,18 +80,43 @@ export interface CowSwapWidgetPalette {
 }
 
 export interface CowSwapWidgetAppParams {
+  /**
+   * Allows to set a custom logo for the widget.
+   */
   logoUrl?: string
+  /**
+   * Option to hide the logo in the widget.
+   */
   hideLogo?: boolean
+  /**
+   * Option to hide the network selector in the widget.
+   */
   hideNetworkSelector?: boolean
+  /**
+   * Enables dynamic height for the widget.
+   * The widget will resize itself to fit the content.
+   */
   dynamicHeightEnabled?: boolean
+  /**
+   * Enables the ability to switch between trade types in the widget.
+   */
   enabledTradeTypes?: TradeType[]
+  /**
+   * Colors palette to customize the widget UI.
+   */
   palette?: CowSwapWidgetPalette
 }
 
 export type CowSwapWidgetSettings = CowSwapWidgetUrlParams & CowSwapWidgetAppParams
 
 export interface CowSwapWidgetMetaData {
```

Comment:
there's some discussions about deleting he metadata, and moving appKey to the main object

Reply:
Already done here:
https://github.com/cowprotocol/cowswap/pull/3318